src/transformers/models/efficientnet/image_processing_efficientnet.py

```diff
@@ -177,26 +185,25 @@ def rescale(
         **kwargs,
     ):
         """
-        Rescale an image by a scale factor. image = image * scale.
+        Rescale an image by a scale factor.
+
+        If offset is `True`, image scaled between [-1, 1]: image = (image - scale / 2) * scale. If offset is `False`,
+        image scaled between [0, 1]: image = image * scale
 
         Args:
             image (`np.ndarray`):
                 Image to rescale.
             scale (`int` or `float`):
                 Scale to apply to the image.
-            offset (`bool`, *optional*):
+           offset (`bool`, *optional*):
                 Whether to scale the image in both negative and positive directions.
             data_format (`str` or `ChannelDimension`, *optional*):
                 The channel dimension format of the image. If not provided, it will be the same as the input image.
         """
+        image = image.astype(np.float32)
```

Comment:
The logic change here is equivalent. It's updated so it's simpler and easier for other models to copy from it.

Reply:
Not against with the change, but it is not copied from `image_transformers.rescale`, and that method doesn't have `image = image.astype(np.float32)`. So I am a bit confused with the above comment.

Reply:
Oh, `rescale` is used in the  return statement! I trust you for the math formula below :-)

Reply:
Actually, the logic for both vivit and efficient net are wrong, as they're not rescaling between [-1, 1]. I'm going to make a separate PR for that fix.